components/identity-event/org.wso2.carbon.identity.event/src/test/java/org.wso2.carbon.identity.event/handler/AbstractEventHandlerTest.java

```diff
@@ -0,0 +1,121 @@
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+
+        String home = IdentityEventConfigBuilder.class.getResource("/").getFile();
+        String config = IdentityEventConfigBuilder.class.getResource("/").getFile();
+        System.setProperty("carbon.home", home);
+        System.setProperty("carbon.config.dir.path", config);
+
+        subscriptionList = new ArrayList<>();
+        moduleConfiguration = Mockito.mock(ModuleConfiguration.class);
+        Mockito.doReturn(subscriptionList).when(moduleConfiguration).getSubscriptions();
+
+        IdentityEventConfigBuilder identityEventConfigBuilder = Mockito.mock(IdentityEventConfigBuilder.class);
+        Field field = IdentityEventConfigBuilder.class.getDeclaredField("notificationMgtConfigBuilder");
+        field.setAccessible(true);
+        field.set(null, identityEventConfigBuilder);
+        Mockito.doReturn(moduleConfiguration).when(identityEventConfigBuilder).getModuleConfigurations(moduleName);
+
+        moduleConfigurationMap = new HashMap<>();
+        Mockito.doReturn(moduleConfigurationMap).when(identityEventConfigBuilder).getModuleConfiguration();
+        Mockito.doAnswer(new Answer() {
+            @Override
+            public Object answer(InvocationOnMock invocationOnMock) throws Throwable {
+                return moduleConfigurationMap.get(invocationOnMock.getArguments()[0]);
+            }
+        }).when(identityEventConfigBuilder).getModuleConfigurations(Matchers.anyString());
+        ModuleConfiguration moduleConfiguration2 = new ModuleConfiguration(new Properties(), subscriptionList);
+        moduleConfigurationMap.put("TestEventHandler", moduleConfiguration2);
+    }
+
+    @Test
+    public void testCanHandle() throws IdentityEventException {
+
+        Event event = new Event("eventName");
+        IdentityEventMessageContext messageContext = new IdentityEventMessageContext(event);
+
+        TestEventHandler testEventHandler = new TestEventHandler();
+        boolean canHandle = testEventHandler.canHandle(messageContext);
+        Assert.assertFalse(canHandle);
+
+        subscriptionList.add(new Subscription("eventName", new Properties()));
+        canHandle = testEventHandler.canHandle(messageContext);
+        Assert.assertTrue(canHandle);
+    }
+
+    @Test
+    public void testIsAssociationAsync() throws IdentityEventException {
+
+        Properties properties = new Properties();
+        subscriptionList.add(new Subscription("eventName", new Properties()));
+        TestEventHandler testEventHandler = new TestEventHandler();
+        boolean isAssociationAsync = testEventHandler.isAssociationAsync("unknownEvent");
+
+        Assert.assertFalse(isAssociationAsync);
+
+        subscriptionList.add(new Subscription("testEvent", new Properties()));
+        isAssociationAsync = testEventHandler.isAssociationAsync("testEvent");
+
+        Assert.assertFalse(isAssociationAsync, "testEvent is not an asynchronous event");
```

Comment:
Assert failure message looks wrong. Should be "testEvent is an asynchronous event" ??